pkg/aws/manager_ec2_security_group.go

```diff
@@ -0,0 +1,114 @@
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/integr8ly/cluster-service/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	loggingKeySecurityGroup = "security-group-id"
+
+	resourceTypeSecurtyGroup = "ec2:security-group"
+)
+
+var _ ClusterResourceManager = &SecurityGroupManager{}
+
+// SecurityGroupManager type
+type SecurityGroupManager struct {
+	ec2Client     ec2Client
+	taggingClient taggingClient
+	logger        *logrus.Entry
+}
+
+// NewDefaultSecurityGroupManager create session for manager
+func NewDefaultSecurityGroupManager(session *session.Session, logger *logrus.Entry) *SecurityGroupManager {
+	return &SecurityGroupManager{
+		ec2Client:     ec2.New(session),
+		taggingClient: resourcegroupstaggingapi.New(session),
+		logger:        logger.WithField(loggingKeyManager, managerSecurityGroup),
+	}
+}
+
+// GetName getter function
+func (r *SecurityGroupManager) GetName() string {
+	return "AWS EC2 SecurityGroup Manager"
+}
+
+// DeleteResourcesForCluster deletes resource for cluster
+func (r *SecurityGroupManager) DeleteResourcesForCluster(clusterId string, tags map[string]string, dryRun bool) ([]*clusterservice.ReportItem, error) {
+	var securityGroupsToDelete []*basicResource
+	r.logger.Debug("delete security groups resources for cluster")
+	//  integreatly.org/clusterID tags
+	resourceInput := &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{resourceTypeSecurtyGroup}),
+		TagFilters:          convertClusterTagsToAWSTagFilter(clusterId, tags),
+	}
+	// add to the security group delete array
+	securityGroupsToDelete, err := r.GetDeleteSecurityGroup(resourceInput, securityGroupsToDelete)
+	if err != nil {
+		r.logger.Errorf("Error:", err)
+	}
+	//delete resources
+	var reportItems []*clusterservice.ReportItem
+	for _, securityGroup := range securityGroupsToDelete {
+		securityGroupLogger := r.logger.WithField(loggingKeySecurityGroup, securityGroup.Name)
+		reportItem := &clusterservice.ReportItem{
+			ID:           securityGroup.ARN,
+			Name:         securityGroup.Name,
+			Action:       clusterservice.ActionDelete,
+			ActionStatus: clusterservice.ActionStatusInProgress,
+		}
+		reportItems = append(reportItems, reportItem)
+		if dryRun {
+			securityGroupLogger.Debugf("dry run is enabled, skipping deletion")
+			reportItem.ActionStatus = clusterservice.ActionStatusDryRun
+			continue
+		}
+		securityGroupLogger.Debugf("performing security group deletion")
+		deleteSecurityGroupInput := &ec2.DeleteSecurityGroupInput{
+			GroupId: aws.String(securityGroup.Name),
+		}
+		if _, err := r.ec2Client.DeleteSecurityGroup(deleteSecurityGroupInput); err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "DependencyViolation" {
+				securityGroupLogger.Debug("security group has existing dependencies which have not been deleted, skipping")
+				reportItem.ActionStatus = clusterservice.ActionStatusSkipped
+				continue
```

Comment:
I am not 100% familiar with this service, so just curious to what happens when we reach here, and the security group can't be deleted, so we continue to try remove it?

Reply:
it will skip , run the clusterservice with the watch command and it will remove it when the condition DependencyViolation is satisfied .

Reply:
Awesome, thanks for the clarification